microprofile/metrics-smallrye/src/main/java/org/wildfly/extension/microprofile/metrics/MicroProfileMetricsSubsystemDefinition.java

```diff
@@ -79,8 +89,43 @@ protected MicroProfileMetricsSubsystemDefinition() {
                 .setCapabilities(METRICS_REGISTRATION_RUNTIME_CAPABILITY, HTTP_CONTEXT_CAPABILITY));
     }
 
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+            super.registerAttributes(resourceRegistration);
+            resourceRegistration.registerReadWriteAttribute(EXPOSED_SUBSYSTEMS, null, new ExposedSubsystemWriterHandler());
+    }
+
     @Override
     public Collection<AttributeDefinition> getAttributes() {
-        return Arrays.asList(ATTRIBUTES);
+        return Arrays.asList(SECURITY_ENABLED);
```

Comment:
That is not correct. this method must return all attributes of the resource. 

If you provide a specific read-write handlers for exposed-subsystems, you also have to register other attributes (such as `security-enabled`) in the `registerAttributes` method.